packages/react/src/select/positioner/SelectPositioner.tsx

```diff
@@ -114,12 +120,58 @@ export const SelectPositioner = React.forwardRef(function SelectPositioner(
       alignItemWithTriggerActive,
       controlledItemAnchor,
       setControlledItemAnchor,
+      scrollUpArrowVisible,
+      setScrollUpArrowVisible,
+      scrollDownArrowVisible,
+      setScrollDownArrowVisible,
     }),
-    [positioner, alignItemWithTriggerActive, controlledItemAnchor],
+    [
+      positioner,
+      alignItemWithTriggerActive,
+      controlledItemAnchor,
+      scrollUpArrowVisible,
+      scrollDownArrowVisible,
+    ],
   );
 
+  const prevMapSize = React.useRef(0);
+
   return (
-    <CompositeList elementsRef={listRef} labelsRef={labelsRef}>
+    <CompositeList
+      elementsRef={listRef}
+      labelsRef={labelsRef}
+      onMapChange={(map) => {
```

Comment:
Use a stable function as `onMapChange` is on `useModernLayoutEffect`'s dependency list in CompositeList.